jupyter_core/paths.py

```diff
@@ -131,6 +132,9 @@ def jupyter_path(*subdirs):
 
     JUPYTER_PATH environment variable has highest priority.
 
+    If the JUPYTER_ENV_PRIORITY environment variable is set, the environment-level
```

Comment:
Just wondering if there might be a case where, say, a server extension might require the current priority model, yet needs to protect itself against the new env, or vice versa (i.e., wants to force the new behavior for its functionality)?

To support that, we could add an optional kwarg (e.g., `env_priority=None`) such that, when set, it takes precedence over the ENV value.  (The argument name should be that of env name sans the `jupyter_` prefix.)

Reply:
I'm having trouble coming up with a case where you want to override the user's explicit wishes that the environment be given priority over the user level. Can you elaborate on this case?

I see this variable as the user giving Jupyter explicit information about a common usecase that was not envisioned when originally coding this path hierarchy.

Reply:
Also, I kind of hesitate to introduce an override of the override - that is getting pretty confusing unless we really have a concrete case for it, I think.

Reply:
Understood. I'm okay with that.  It's something that could be added at any time (following the existence of a concrete case).